formats/tabular-converter-plugin/src/polus/plugins/formats/tabular_converter/tabular_converter.py

```diff
@@ -0,0 +1,180 @@
+
+
+class Extension(str, enum.Enum):
+    """Extension types to be converted."""
+
+    FITS = ".fits"
+    FEATHER = ".feather"
+    PARQUET = ".parquet"
+    HDF = ".hdf5"
+    FCS = ".fcs"
+    CSV = ".csv"
+    ARROW = ".arrow"
+    Default = "default"
+
+
+class Convert_tabular:
+    """Convert vaex supported file formats into Arrow data format and vice versa.
+
+    Args:
+        file: Path to input file.
+        file_pattern : Desired ouput file extension.
+        out_dir: Path to save the output csv file.
+    """
+
+    def __init__(self, file: pathlib.Path, file_pattern: str, out_dir: pathlib.Path):
+        """Define Instance attributes."""
+        self.file = file
+        self.out_dir = out_dir
+        self.file_pattern = file_pattern
+        self.file_pattern = self.get_file_pattern()
+        self.POLUS_TAB_EXT = (
+            POLUS_TAB_EXT if self.file_pattern is None else self.file_pattern
+        )
+        self.output_file = pathlib.Path(
+            self.out_dir, (self.file.stem + self.POLUS_TAB_EXT)
+        )
+
+    def get_file_pattern(self) -> str:
+        """Define file_patterns."""
+        if self.file_pattern == Extension.Default:
+            file_pattern = POLUS_TAB_EXT
+        elif self.file_pattern == Extension.FITS:
+            file_pattern = ".fits"
+        elif self.file_pattern == Extension.FEATHER:
+            file_pattern = ".feather"
+        elif self.file_pattern == Extension.PARQUET:
+            file_pattern = ".parquet"
+        elif self.file_pattern == Extension.HDF:
+            file_pattern = ".hdf5"
+        elif self.file_pattern == Extension.FCS:
+            file_pattern = ".fcs"
+        elif self.file_pattern == Extension.CSV:
+            file_pattern = ".csv"
+        elif self.file_pattern == Extension.ARROW:
+            file_pattern = ".arrow"
+        return file_pattern
+
+    def csv_to_df(self) -> vaex.DataFrame:
+        """Convert csv into datafram or hdf5 file."""
+        logger.info("csv_to_df: Copy csv file into out_dir for processing...")
+        logger.info("csv_to_df: Checking size of csv file...")
+        # Open csv file and count rows in file
+        with open(self.file, encoding="utf-8") as fr:
+            ncols = len(fr.readline().split(","))
+        chunk_size = max([2**24 // ncols, 1])
+        logger.info("csv_to_df: # of columns are: " + str(ncols))
+        # Convert large csv files to hdf5 if more than 1,000,000 rows
+        logger.info("csv_to_df: converting file into hdf5 format")
+        df = vaex.from_csv(self.file, convert=True, chunk_size=chunk_size)
+        return df
+
+    def binary_to_df(self) -> vaex.DataFrame:
+        """Convert any binary formats into vaex dataframe."""
+        binary_patterns = [".fits", ".feather", ".parquet", ".hdf5", ".arrow"]
+        logger.info("binary_to_df: Scanning directory for binary file pattern... ")
+        if self.file_pattern in binary_patterns:
```

Comment:
Once the type of `self.file_pattern` is `Extension` instead of `str`, this `if` check will become unnecessary.